addons/website_mass_mailing/controllers/main.py

```diff
@@ -40,14 +40,29 @@ def subscribe(self, list_id, value, subscription_type, **post):
                 'toast_content': _("Suspicious activity detected by Google reCaptcha."),
             }
 
+        success_toast = {
+            'toast_type': 'success',
+            'toast_content': _("Thanks for subscribing!"),
+        }
+
         ContactSubscription = request.env['mailing.contact.subscription'].sudo()
         Contacts = request.env['mailing.contact'].sudo()
+        MailingList = request.env['mailing.list'].sudo()
+
         if subscription_type == 'email':
             name, value = Contacts.get_name_email(value)
         elif subscription_type == 'mobile':
             name = value
 
         fname = self._get_fname(subscription_type)
+
+        if not list_id or not MailingList.search([('id', '=', list_id)]):
```

Comment:
shouldn't we return an error instead ? (it probably means that the website is not up to date with the backend)

otherwise it might be hard to notice that our website is referring an old mailing list

Reply:
Yes, I brought this up to lna and std. We concluded that since it's on the side of the portal users, it's better not to show them messages that aren't relevant to them. (Since that particular mailing list doesn't exist anymore, we probably don't have anything interesting to send to them anyway)

Originally I had a warning toast telling them the mailing list was discontinued, but lna didn't want that either.

Reply:
@std-odoo Hey, concerning this actually. 

lna asked if I could add a warning for the users of the website on these lists that don't exist anymore. 

Could you check out the top-most commit ('temp') here. I used a related, existing, RPC to try and avoid having one just for that. And I insert the HTML directly since it should barely happen at all, and the way the snippets work uses static python template at generation so I can't use a dynamic t-if or anything fancy.

Thanks in advance :)